fastly/base_fastly_service_v1.go

```diff
@@ -0,0 +1,482 @@
+		Type:    serviceDef.GetType(),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(service.ID)
+	return resourceServiceUpdate(d, meta, serviceDef)
+}
+
+// resourceServiceUpdate provides service resource Update functionality
+func resourceServiceUpdate(d *schema.ResourceData, meta interface{}, serviceDef ServiceDefinition) error {
+	if err := validateVCLs(d); err != nil {
+		return err
+	}
+
+	conn := meta.(*FastlyClient).conn
+
+	// Update Name and/or Comment. No new verions is required for this
+	if d.HasChange("name") || d.HasChange("comment") {
+		_, err := conn.UpdateService(&gofastly.UpdateServiceInput{
+			ID:      d.Id(),
+			Name:    d.Get("name").(string),
+			Comment: d.Get("comment").(string),
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Once activated, Versions are locked and become immutable. This is true for
+	// versions that are no longer active. For Domains, Backends, DefaultHost and
+	// DefaultTTL, a new Version must be created first, and updates posted to that
+	// Version. Loop these attributes and determine if we need to create a new version first
+	var needsChange bool
+	for _, a := range serviceDef.GetAttributeHandler() {
+		if a.HasChange(d) {
+			needsChange = true
+			break
+		}
+	}
+
+	// Update the active version's comment. No new version is required for this
+	if d.HasChange("version_comment") && !needsChange {
+		latestVersion := d.Get("active_version").(int)
+		if latestVersion == 0 {
+			// If the service was just created, there is an empty Version 1 available
+			// that is unlocked and can be updated
+			latestVersion = 1
+		}
+
+		opts := gofastly.UpdateVersionInput{
+			Service: d.Id(),
+			Version: latestVersion,
+			Comment: d.Get("version_comment").(string),
+		}
+
+		log.Printf("[DEBUG] Update Version opts: %#v", opts)
+		_, err := conn.UpdateVersion(&opts)
+		if err != nil {
+			return err
+		}
+	}
+
+	initialVersion := false
+
+	if needsChange {
+		latestVersion := d.Get("active_version").(int)
+		if latestVersion == 0 {
+			initialVersion = true
+			// If the service was just created, there is an empty Version 1 available
+			// that is unlocked and can be updated
+			latestVersion = 1
+		} else {
+			// Clone the latest version, giving us an unlocked version we can modify
+			log.Printf("[DEBUG] Creating clone of version (%d) for updates", latestVersion)
+			newVersion, err := conn.CloneVersion(&gofastly.CloneVersionInput{
+				Service: d.Id(),
+				Version: latestVersion,
+			})
+			if err != nil {
+				return err
+			}
+
+			// The new version number is named "Number", but it's actually a string
+			latestVersion = newVersion.Number
+			d.Set("cloned_version", latestVersion)
+
+			// New versions are not immediately found in the API, or are not
+			// immediately mutable, so we need to sleep a few and let Fastly ready
+			// itself. Typically, 7 seconds is enough
+			log.Print("[DEBUG] Sleeping 7 seconds to allow Fastly Version to be available")
+			time.Sleep(7 * time.Second)
```

Comment:
@thommahoney not blocking for this PR, but I'd like to understand why this is the case.